auth/src/main/java/org/aerogear/auth/impl/AbstractAuthenticator.java

```diff
@@ -0,0 +1,32 @@
+package org.aerogear.auth.impl;
+
+import org.aerogear.auth.AuthenticationException;
+import org.aerogear.auth.credentials.ICredential;
+
+import java.security.Principal;
+
+/**
+ * Base class for all authenticators
+ */
+public class AbstractAuthenticator {
+
+    /**
+     * This method must be overridden with the custom authentication for the given credential.
+     *
+     * @param username username to be authenticated
+     * @param credential user credential
+     * @return the authenticated principal
+     * @throws AuthenticationException
+     */
+    public Principal authenticate(final String username, ICredential credential) throws AuthenticationException {
```

Comment:
The only things we will need to pass in here is some auth config. This will tell the appauth library where the keycloak server is, what realm and client we want to communicate under, what OpenID Connect scopes we want to pass, what browsers we want the users to use to authenticate etc. We don't need to pass any credentials here.